src/java.base/share/classes/java/io/DataOutputStream.java

```diff
@@ -377,25 +379,11 @@ static int writeUTF(String str, DataOutput out) throws IOException {
         int count = 0;
         ByteArray.setUnsignedShort(bytearr, count, utflen);
         count += 2;
-        int i = 0;
-        for (i = 0; i < strlen; i++) { // optimized for initial run of ASCII
-            int c = str.charAt(i);
-            if (c >= 0x80 || c == 0) break;
-            bytearr[count++] = (byte) c;
-        }
+        str.getBytes(0, countNonZeroAscii, bytearr, count);
```

Comment:
Maybe the deprecated [`String​::getBytes​(int, int, byte[], int)`] method calls could be moved to a method in `JDKUTF` in order to avoid having to use overly broad `@SuppressWarnings("deprecation")`:
```java
@ForceInline
@SuppressWarnings("deprecation")
public static void getBytes(String str, int srcPos, byte[] dst, int dstPos, int length) {
	str.getBytes(srcPos, srcPos + length, dst, dstPos);
}
```

[`String​::getBytes​(int, int, byte[], int)`]: https://docs.oracle.com/en/java/javase/22/docs/api/java.base/java/lang/String.html#getBytes(int,int,byte[],int)

Reply:
Although this will eliminate the warning, JDKUTF provides capabilities that are unrelated to it. What happens if a scenario other than JDKUTF is called?